cmake/Modules/CompilerConfig.cmake

```diff
@@ -84,16 +84,21 @@ else()
   endif()
 
   add_compile_options(
+    -Werror
     -Wextra
     -Wvla
+    -Wformat
+    -Wformat-security
     -Wno-unused-function
     -Wno-missing-field-initializers
+    -Wno-error=deprecated-declarations
```

Comment:
I'd rather add this to projects that we _know_ use deprecated technology (e.g. libobs-opengl) and still keep this as an error otherwise.

We need to ensure that new code doesn't use deprecated APIs without anyone noticing.

Reply:
I remembered why I put it there because of Qt6 deprecation warnings, actually testing to apply the flag to each "library/binary" that use Qt.